newIDE/app/src/AssetStore/BehaviorStore/BehaviorStoreContext.js

```diff
@@ -0,0 +1,292 @@
+            error
+          );
+          setError(error);
+        }
+
+        isLoading.current = false;
+      })();
+    },
+    [behaviorShortHeadersByType, isLoading]
+  );
+
+  React.useEffect(
+    () => {
+      // Don't attempt to load again extensions and filters if they
+      // were loaded already.
+      if (Object.keys(behaviorShortHeadersByType).length || isLoading.current)
+        return;
+
+      const timeoutId = setTimeout(() => {
+        console.info('Pre-fetching behaviors from extension store...');
+        fetchBehaviorsAndFilters();
+      }, 5000);
+      return () => clearTimeout(timeoutId);
+    },
+    [fetchBehaviorsAndFilters, behaviorShortHeadersByType, isLoading]
+  );
+
+  const allBehaviors = React.useMemo(
+    () => {
+      /** @type {{[name: string]: BehaviorShortHeader | SearchableBehaviorMetadata}} */
+      const allBehaviors = {};
+      for (const type in behaviorShortHeadersByType) {
+        allBehaviors[type] = behaviorShortHeadersByType[type];
+      }
+      for (const installedBehaviorMetadata of installedBehaviorMetadataList) {
+        allBehaviors[
+          installedBehaviorMetadata.type
+        ] = installedBehaviorMetadata;
+      }
+      return allBehaviors;
+    },
+    [installedBehaviorMetadataList, behaviorShortHeadersByType]
+  );
+
+  const allCategories = React.useMemo(
+    () => {
+      const categoriesSet = new Set();
+      for (const type in allBehaviors) {
+        categoriesSet.add(allBehaviors[type].category);
+      }
+      const sortedCategories = [...categoriesSet].sort((tag1, tag2) =>
+        tag1.toLowerCase().localeCompare(tag2.toLowerCase())
+      );
+      return sortedCategories;
+    },
+    [allBehaviors]
+  );
+
+  const filters = React.useMemo(
+    () => {
+      const tagsSet = new Set();
+      for (const type in allBehaviors) {
+        allBehaviors[type].tags.forEach(tag => tagsSet.add(tag));
+      }
+      const sortedTags = [...tagsSet].sort((tag1, tag2) =>
+        tag1.toLowerCase().localeCompare(tag2.toLowerCase())
+      );
+      return {
+        allTags: sortedTags,
+        defaultTags: sortedTags,
+        tagsTree: [],
+      };
+    },
+    [allBehaviors]
+  );
+
+  const defaultFirstSearchItemIds = React.useMemo(
+    () => [
+      ...installedBehaviorMetadataList.map(behavior => behavior.type),
+      ...firstBehaviorIds,
+    ],
+    [firstBehaviorIds, installedBehaviorMetadataList]
+  );
+
+  const searchResults: ?Array<{|
+    item: BehaviorShortHeader | SearchableBehaviorMetadata,
+    matches: SearchMatch[],
+  |}> = useSearchStructuredItem(allBehaviors, {
+    searchText,
+    chosenItemCategory: chosenCategory,
+    chosenCategory: filtersState.chosenCategory,
+    chosenFilters: filtersState.chosenFilters,
+    excludedTiers: showCommunityExtensions
+      ? noExcludedTiers
+      : excludedCommunityTiers,
+    defaultFirstSearchItemIds: defaultFirstSearchItemIds,
+  });
+
+  const hasBehaviorNamed = React.useCallback(
```

Comment:
What is this used for?

Reply:
It's a leftover from refactoring. For the extensions, it's used to check if an unknown instruction comes from a store extension.